core/integration/container_test.go

```diff
@@ -3556,3 +3556,65 @@ func TestContainerParallelMutation(t *testing.T) {
 	require.Equal(t, res.Container.A.EnvVariable, "BAR")
 	require.Empty(t, res.Container.B, "BAR")
 }
+
+func TestContainerBuildConsistency(t *testing.T) {
```

Comment:
nit: I probably won't understand what this test name means months from now (consistent with what?) :)

```suggestion
func TestContainerBuildMergesWithParent(t *testing.T) {
```